java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java

```diff
@@ -840,36 +847,29 @@ public ResponseEntity<String> generateGpxSharedUrl(@RequestParam String name,
 		if (userFile == null) {
 			return ResponseEntity.badRequest().body(FILE_NOT_FOUND);
 		}
-		String sharedUrl = userdataService.generateSharedUrl(userFile);
+		String code = shareGpxService.generateSharedCode(userFile);
+		String domain = request.getScheme() + "://" + request.getServerName() +
+				(request.getServerPort() != 80 && request.getServerPort() != 443 ? ":" + request.getServerPort() : "");
```

Comment:
It's better to not use OsmAnd Server Name/Port in the URL. The URL should exclude domain/port and should be processed on the Client (Web - relative to current site URL, Android - fixed osmand.net/..., etc).